tests/unit/synapseclient/unit_test_commandline.py

```diff
@@ -191,6 +195,42 @@ def login_side_effect(*args, **kwargs):
         login.side_effect = login_side_effect
 
         cmdline._authenticate_login(syn, username, password, **login_kwargs)
+        assert expected_success_login_calls == login.call_args_list
+
+
+def test_authenticate_login__auth_token(syn):
+    """Verify attempting to authenticate when supplying an auth bearer token instead of an password (or api key).
+    Should attempt to treat the password as an api key after the initial failure as a password and api key"""
+
+    username = 'foo'
+    auth_token = 'auth_bearer_token'
+    login_kwargs = {'rememberMe': True}
+
+    # since this token does not appear to be base64 encoded it is not tried as an api key
+    expected_login_calls = [
+        call(username, password=auth_token, **login_kwargs),
+        call(username, authToken=auth_token, **login_kwargs),
+    ]
+
+    with patch.object(syn, 'login') as login:
+        login.side_effect = SynapseAuthenticationError()
+
+        # simulate failure both as password and as api key
+        with pytest.raises(SynapseAuthenticationError):
+            cmdline._authenticate_login(syn, username, auth_token, **login_kwargs)
+
+        assert expected_login_calls == login.call_args_list
+        login.reset_mock()
+
+        # now simulate success when used as an api key
```

Comment:
```suggestion
        # now simulate success when used as an auth token
```

Reply:
If we do change the order such that auth_token > api_key, then the test for api_token should have a simulated auth_token failure